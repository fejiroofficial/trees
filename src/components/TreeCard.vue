<template>
  <div class="tree-card ant-card ant-card-bordered">
    <div class="tree-card__name">{{ tree.name }}</div>
    <p>{{ tree.species_name }}</p>
    <img
      v-if="showImage"
      :alt="tree.name"
      :src="tree.image"
      style="width: 200px; height: 200px"
    />
    <button class="ant-btn" @click="showImage = !showImage" style="margin-top: 10px;">{{
      showImage ? "Hide Image" : "Show Image"
    }}</button>
  </div>
</template>

<script>
export default {
  props: {
    tree: Object,
  },
  data() {
    return {
      showImage: false,
    };
  },
};
</script>

<style lang="scss" scoped>
.tree-card {
  width: 300px;
  margin-bottom: 50px;
  padding: 20px;

  &__name {
    font-weight: bold;
    font-size: x-large;
  }
}
.ant-card {
  margin: 20px;
}
</style>
