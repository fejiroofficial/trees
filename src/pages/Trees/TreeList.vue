<template>
  <div>
    <h1>Tree List</h1>
  </div>
</template>
