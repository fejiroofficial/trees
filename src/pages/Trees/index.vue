<template>
  <div>
    <h1>Trees</h1>
  </div>
</template>
