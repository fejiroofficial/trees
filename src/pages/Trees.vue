<template>
  <div class="trees-container">
    <Logo />
    <p>"On the last day of the world I would want to plant a tree."</p>
    <p>â€• W.S. Merwin</p>
    <div class="trees-container__display">
      <div class="trees-container__display-inner" v-if="trees">
        <tree-card :tree="tree" v-for="tree in trees" :key="tree.name" />
      </div>
    </div>
  </div>
</template>

<script>
import Logo from "@/components/Logo";
import TreeCard from "@/components/TreeCard";

export default {
  components: {
    Logo,
    TreeCard,
  },
  created() {
    this.$store.dispatch("trees/fetchTrees");
  },
  computed: {
    trees() {
      return this.$store.getters["trees/getTrees"];
    },
  },
};
</script>

<style lang="scss" scoped>
.trees-container {
  text-align: center;
  &__display {
    display: grid;
    justify-content: center;
  }
  &__display-inner {
    width: 43rem;
    display: flex;
    flex-flow: wrap;

    @media (max-width: 675px) {
      width: 20rem;
    }
  }
}
</style>
