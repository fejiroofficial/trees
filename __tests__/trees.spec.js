import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Trees from "@/pages/Trees";
import TreeCard from "@/components/TreeCard";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Trees.vue", () => {
  let wrapper;
  let actions;
  let getters;
  let store;

  beforeEach(() => {
    actions = {
      fetchTrees: jest.fn(),
    };
    getters = {
      getTrees: () => [
        {
          name: "sample tree",
          species_name: "sample species",
          image: "www.image.com",
        },
      ],
    };
    store = new Vuex.Store({
      actions,
      getters,
    });
    wrapper = shallowMount(Trees, {
      store,
      localVue,
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("dispatches get trees as component is created", () => {
    expect(actions.fetchTrees).toHaveBeenCalled();
  });

  it("gets trees from store to display", () => {
    expect(wrapper.vm.trees).toStrictEqual(getters.getTrees());
  });

  it("mount tree cards", () => {
    expect(wrapper.findComponent(TreeCard).exists()).toBe(true);
  });
});
