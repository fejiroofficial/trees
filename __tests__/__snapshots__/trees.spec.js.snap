// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trees.vue renders correctly 1`] = `
<div
  class="trees-container"
>
  <logo-stub />
   
  <p>
    "On the last day of the world I would want to plant a tree."
  </p>
   
  <p>
    â€• W.S. Merwin
  </p>
   
  <div
    class="trees-container__display"
  >
    <div
      class="trees-container__display-inner"
    >
      <tree-card-stub
        tree="[object Object]"
      />
    </div>
  </div>
</div>
`;
